recipe_app/models.py

```diff
@@ -0,0 +1,41 @@
+from typing import Dict, Any
+import emoji
+
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+from django.core.exceptions import ValidationError
+
+
+def validate_icon(icon: str):
+    """
+    Check if a given string is a single, valid emoji.
+
+    Args:
+      icon (str): The `icon` parameter is a string that represents an emoji icon.
+
+    Returns:
+        True if the string is a valid emoji, otherwise raises a ValidationError.
+    """
+    if emoji.is_emoji(icon) is False:
+        raise ValidationError(
+            _("%(icon)s is not a valid emoji"),
+            params={"icon": icon},
+        )
+
+    return True
+
+
+# Create your models here.
+class Ingredient(models.Model):
+    id = models.AutoField(primary_key=True)
+    name = models.CharField(max_length=255)
+    recipe = models.ForeignKey(
+        "Recipe", related_name="ingredients", on_delete=models.CASCADE
+    )
+
+
+class Recipe(models.Model):
+    id = models.AutoField(primary_key=True)
```

Comment:
I know they are incredibly small, but I would create the class/model in a separate file 😄  and move the `validate_icon` to a private method for its own class or its own space.

Reply:
Yep I agree, I've moved the function to a new `validators.py` file for consistency